### PR TITLE
After reaching quorum, wait for additional validators.

### PR DESCRIPTION
**Pull request by afck**
This introduces a short grace period, proportional to how long it took to reach a quorum, to allow additional validators to contribute signatures or chain information.

One place where this is useful is `send_certificate`: When updating validators, we re-send the full certificate to everyone whose signature isn't in it. With the grace period, often every validator will have signed it, and only `LiteCertificate`s need to be sent.

This also fixes a bug: We need to recompute the total and thresholds when adding and removing validators.

### CODE REVIEW

File: linera-core/src/client.rs [hunk 1]
```diff
@@ -1183,8 +1183,7 @@ where
 
     /// Attempts to update all validators about the local chain.
     pub async fn update_validators(&mut self) -> Result<()> {
-        let mut committee = self.local_committee().await?;
-        committee.quorum_threshold = committee.total_votes;
```
Comment (afck) [inline linera-core/src/client.rs:1187]: Maybe the grace period achieves a similar effect?

File: linera-execution/src/committee.rs [hunk 2]
```diff
@@ -155,6 +155,40 @@ impl Committee {
...
+                return false; // No change.
+            }
+            new_total -= state.votes;
+        }
+        if let Some(new_total) = new_total.checked_add(new_state.votes) {
+            self.total_votes = new_total;
+        } else {
+            return false;
+        };
+        self.validators.insert(name, new_state);
```
Comment (jvff) [inline linera-execution/src/committee.rs:179]: Question: don't the `*_threshold`s have to be updated here as well?

Comment (afck) [reply linera-execution/src/committee.rs]: Ah, sorry, I was meaning to delete this method again! (I forgot to save. :facepalm:)
I pushed the fixed commit.

**Review by ma2bd — APPROVED** [review]

File: linera-core/src/updater.rs [hunk 3]
```diff
@@ -19,10 +19,16 @@ use std::{
...
-    time::Duration,
+    time::{Duration, Instant},
 };
 use tracing::{error, info, warn};
 
+/// The amount of time we wait for additional validators to contribute to the result, as a fraction
+/// of how long it took to reach a quorum.
+const GRACE_PERIOD: f64 = 0.2;
+/// The maximum timeout for `communicate_with_quorum` if no quorum is reached.
+const MAX_TIMEOUT: Duration = Duration::from_secs(60 * 60 * 24); // 1 day.
```
[the remark this answers is not in the document]
Comment (afck) [reply linera-core/src/updater.rs:30]: I just needed some value for when no `end_time` was computed yet. I don't think there's an `Instant::MAX`, and I imagine adding `Duration::MAX` to the current time would overflow.

**Pull request merged**